darts/timeseries.py

```diff
@@ -293,6 +292,10 @@ def __init__(self, xa: xr.DataArray):
         # Store static covariates and hierarchy in attributes (potentially storing None)
         self._xa = _xarray_with_attrs(self._xa, static_covariates, hierarchy)
 
+        # at the end since other exceptions can prevent the creation of the time serie
+        if not xa.size > 0:
+            logger.warning("TimeSeries is empty.")
```

Comment:
Actually, I think we maybe shouldn't warn. I think the main reason for allowing empty series is to allow more general kinds of "time series arithmetic", which I would guess happen typically when we mass-process lots of series, and it might be a pain to have lots of warnings.

Reply:
I was just worried that some users could end up with weird issues (during training for example) and eventually struggle to identify the source of the issue but I did not take the time to check what kind of errors happen when these methods are called with an empty TimeSeries (I could eventually add exceptions/warning in these methods instead of the constructor?). It's true that when handling massive amount of TimeSeries, the warnings could become quite annoying.